python/SWESimulators/CDKLM16.py

```diff
@@ -267,8 +270,8 @@ def step(self, t_end=0.0, apply_stochastic_term=True):
         n = int(t_end / self.dt + 1)
 
         if self.t == 0:
-            self.bc_kernel.boundaryCondition(self.cl_queue, \
-                self.cl_data.h0, self.cl_data.hu0, self.cl_data.hv0)
+            self.bc_kernel.boundaryCondition(self.gpu_stream, \
```

Comment:
Here, streams are being used. I suggest that all GPU objects (kernels, memory, etc.) get a stream as a mandatory argument.

Reply:
That might be the best solution. I propose to do that change in another PR.